app/option.go

```diff
@@ -0,0 +1,24 @@
+package app
+
+import (
+	"net/http"
+)
+
+// ServerOption is an interface to Options that can be used to configure a
+// http server configured behavior. Ready-to-use Option can be found in
+// the `appoptions` package.
+type ServerOption func(*http.Server) error
+
+// SetupServer will apply @options on @server, in the order that they
+// are declared. If an option returns an error, this function will immediately
+// return the received error.
+func SetupServer(server *http.Server, options ...ServerOption) error {
```

Comment:
How about taking this file, along with app/appoptions/appoptions.go and create a `httputil` or `httpserver` package?
It would be used like `server, err := httpserver.New()` or `server, err := httpserver.New(httpserver.WithTimeout(t))`